core/src/main/java/org/jruby/util/RubyFiletypeDetector.java

```diff
@@ -43,6 +43,9 @@ public String probeContentType(Path path) throws IOException {
             if (firstLine != null && SHEBANG_REGEXP.matcher(firstLine).matches()) {
                 return RUBY_MIME;
             }
+        } catch (Exception e) {
+            // Reading random files as UTF-8 could cause all sorts of errors
```

Comment:
Apparently you can throw an `IOException`. But this confuses me - it looks like the code that calls this method doesn't catch any exceptions, so an error in any of your file type detectors causes any further detectors to fail. This makes our detector robust.